openmls/src/group/mls_group/updates.rs

```diff
@@ -33,10 +36,20 @@ impl MlsGroup {
 
         // Create Commit over all proposals. If a `KeyPackageBundle` was passed
         // in, use it to create an update proposal by value. TODO #751
-        let create_commit_result = match key_package_bundle_option {
-            Some(kpb) => {
+        let create_commit_result = match key_pair {
+            Some(key_pair) => {
+                let group_id = self.group_id().clone();
+                let own_leaf = self.group.treesync_mut().own_leaf_node_mut().map_err(|_| {
```

Comment:
As noted at the `own_leaf_node_mut` definition, this might be problematic. In particular, here we change the tree before the commit has been merged. I'm actually curious why this doesn't lead to problems when a commit is created but not merged. But maybe I'm overlooking something.